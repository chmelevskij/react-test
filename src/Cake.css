.Cake {
  position: relative;
  padding: 10px;
  margin: 10px;
  background: rgba(255, 255, 255, 0.5);
  border-radius: 5px;
}

.Cake-remove {
  position: absolute;
  color: white;
  background: #d32f2f;
  right: 5px;
  top: 5px;
  font-size: 1.5rem;
  border-radius: 50px;
  height: 50px;
  width: 50px;
}

.Cake-title {
 text-transform: capitalize;
}

.Cake-image {
  width: 100%;
}
