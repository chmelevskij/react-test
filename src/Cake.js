import React from 'react'

const Cake = ({title, desc, image, cakeId, onClick}) => (
  <li>
    <h2> {title} </h2>
    <p> {desc} </p>
    <img src={image} alt={title}/>
    <button onClick={onClick} value={cakeId}>X</button>
  </li>
)

export default Cake
