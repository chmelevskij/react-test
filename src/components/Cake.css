.Cake {
  position: relative;
  padding: 10px;
  margin: 10px;
  background: rgba(255, 255, 255, 0.5);
  border-radius: 5px;
}

.Cake-remove,
.Cake-finish,
.Cake-edit {
  position: absolute;
  color: white;
  background: #d32f2f;
  right: 5px;
  top: 5px;
  font-size: 1.5rem;
  border-radius: 50px;
  height: 50px;
  width: 50px;
}

.Cake-edit {
  right: 65px;
  background: #1976D2;
}

.Cake-finish {
  right: 65px;
  background: #388e3c;
}

.Cake-title {
 text-transform: capitalize;
}

.Cake-image {
  width: 100%;
}
