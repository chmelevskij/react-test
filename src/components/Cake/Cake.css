.Cake {
  position: relative;
  padding: 10px;
  margin: 10px;
  background: rgba(255, 255, 255, 0.5);
  border-radius: 5px;
}

.Cake-remove,
.Cake-finish,
.Cake-edit {
  position: absolute;
  color: white;
  background: #d32f2f;
  right: 25px;
  bottom: 25px;
  font-size: 1.5rem;
  border-radius: 50px;
  height: 50px;
  width: 50px;
}

.Cake-edit {
  right: 85px;
  background: #1976D2;
}

.Cake-finish {
  right: 85px;
  background: #388e3c;
}

.Cake-title {
 text-transform: capitalize;
}

.Cake-image {
  width: 100%;
}
