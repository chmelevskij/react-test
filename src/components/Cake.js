import React, {Component} from 'react'
import CakeForm from './CakeForm.js'
import './Cake.css'

class Cake extends Component {
  constructor({image, title, desc, cakeId, onCakeUpdate}){
    super()
    this.state = {
      title,
      desc,
      cakeId,
      image,
      editable: false,
      onCakeUpdate
    }

    this.finishEdit = this.finishEdit.bind(this)
    this.edit = this.edit.bind(this)
    this.handleInput = this.handleInput.bind(this)
  }

  finishEdit(event){
    let {title, desc, cakeId, onCakeUpdate, editable} = this.state

    onCakeUpdate(cakeId, title, desc)
    this.setState({editable: !editable})
  }

  handleInput(event){
    this.setState({
      [event.target.name]: event.target.value
    }) 
  }

  componentWillReceiveProps(nextProps){
    let {title, desc, cakeId, image} = nextProps
    this.setState({title, desc, cakeId, image})
  }

  edit(){
    this.setState({editable: !this.state.editable})
  }

  render(){
    let {cakeId, onRemove} = this.props
    let {image, editable ,title, desc} = this.state

    return (
      <li className="Cake">
        <div style={{display: editable ? "none" : "initial"}}>
          <h2 className="Cake-title"> {title} </h2>
          <p> {desc} </p>
          <img className="Cake-image" src={image} alt={title}/>
        </div>
        <CakeForm {...this.state} visible={editable} handleInput={this.handleInput} onSubmit={this.finishEdit} />
        <button className="Cake-finish" onClick={this.finishEdit} style={{ display: editable ? "initial" : "none" }} value={cakeId}>&#128077;</button>
        <button className="Cake-edit" onClick={this.edit} style={{ display: editable ? "none" : "initial" }} value={cakeId}>&#128394;</button>
        <button className="Cake-remove" onClick={onRemove} value={cakeId}>&#128465;</button>
      </li>
    )
  }
}

export default Cake
