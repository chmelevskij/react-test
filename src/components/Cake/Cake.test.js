import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Cake from './Cake.js'

describe('<Cake />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Cake />, div)
  })

  it('Renders Cake', () => {
    let wrapper = renderer.create(<Cake />).toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})
