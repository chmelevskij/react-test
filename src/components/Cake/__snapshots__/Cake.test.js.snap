// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cake /> Renders Cake 1`] = `
<li
  className="Cake"
>
  <div
    style={
      Object {
        "display": "initial",
      }
    }
  >
    <h2
      className="Cake-title"
    >
       
       
    </h2>
    <p>
       
       
    </p>
    <img
      alt={undefined}
      className="Cake-image"
      src={undefined}
    />
  </div>
  <form
    className="CakeForm"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <label
      htmlFor="title"
    >
       Title 
    </label>
    <input
      className="CakeForm-input"
      id="title"
      name="title"
      onChange={[Function]}
      placeholder="Title"
      required={true}
      type="text"
      value={undefined}
    />
    <label
      htmlFor="desc"
    >
       Description 
    </label>
    <textarea
      className="CakeForm-input"
      cols="30"
      id="desc"
      name="desc"
      onChange={[Function]}
      placeholder="Description"
      rows="10"
      value={undefined}
    />
    <label
      htmlFor="image"
    >
       Image 
    </label>
    <input
      className="CakeForm-input"
      id="image"
      name="image"
      onChange={[Function]}
      placeholder="Image URL"
      type="url"
      value={undefined}
    />
    <button
      className="CakeForm-submit CakeForm-input"
      onClick={[Function]}
      style={
        Object {
          "display": "initial",
        }
      }
      type="submit"
    >
      Add
    </button>
  </form>
  <div>
    <button
      className="Cake-finish"
      onClick={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      value={undefined}
    >
       👍 
    </button>
    <button
      className="Cake-edit"
      onClick={[Function]}
      style={
        Object {
          "display": "initial",
        }
      }
      value={undefined}
    >
       🖊 
    </button>
    <button
      className="Cake-remove"
      onClick={undefined}
      style={
        Object {
          "display": "initial",
        }
      }
      value={undefined}
    >
       🗑 
    </button>
  </div>
</li>
`;
